storage/tianmu/core/engine_execute.cpp

```diff
@@ -456,12 +456,14 @@ QueryRouteTo Engine::Execute(THD *thd, LEX *lex, Query_result *result_output, SE
       std::string table_path = Engine::GetTablePath(((Query_tables_list *)lex)->query_tables->table);
       rct = current_txn_->GetTableByPathIfExists(table_path);
     }
-    if (unit_for_union != nullptr && !unit_for_union->is_prepared()) {
-      int res = result_output->prepare(unit_for_union->item_list, unit_for_union);
-      if (res) {
-        TIANMU_LOG(LogCtl_Level::ERROR, "Error: Unsupported UNION");
-        my_message(ER_UNKNOWN_ERROR, "Tianmu: unsupported UNION", MYF(0));
-        throw ReturnMeToMySQLWithError();
+    if (unit_for_union != nullptr) {
+      if (lex->sql_command != SQLCOM_CREATE_TABLE) {  //  for exclude CTAS
```

Comment:
1: Why do we exclude CTAS from this.
2: can be rewrite as `if (unit_for_union != nullptr && lex->sql_command != SQLCOM_CREATE_TABLE )`

Reply:
if not exclude CATS from this, db will crash, as below sql ,from mtr:issue.226:
```
 CREATE TABLE t1 (f1 VARCHAR(255) CHARACTER SET utf8) ENGINE=TIANMU;
 CREATE TABLE t2 AS SELECT LEFT(f1,171) AS f2 FROM t1 UNION SELECT LEFT(f1,171) AS f2 FROM t1;
```

Reply:
The purpose of this writing is to create a correct table structure from union select

Reply:
have rewrited